# Symétries du cristal

In [2]:
from pymatgen.ext.matproj import MPRester
from pymatgen.io.cif import CifWriter
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.core.structure import Molecule
from pymatgen.symmetry.groups import SpaceGroup
from pymatgen.core.operations import SymmOp


# Permet d'utiliser l'interface du MP
with MPRester("2r9TvKUOgDsTx87F")as mpr:
    
    # Permet d'obtenir l'objet 'structure' sur le MP
    structure = mpr.get_structure_by_material_id("mp-568690")
    Sym=SpacegroupAnalyzer(structure)
    new_s=Sym.get_primitive_standard_structure()
    
    # Crée un objet permettant de convertir la structure en fichier cif
    cwsym=CifWriter(structure,0.1) # Maille hexagonale
    cwsym.write_file("CdBr2sym")
    cwprim=CifWriter(new_s)     #Maille primitive
    cwprim.write_file("CdBr2prim")
    
    

## Groupe d'espace associé au CdBr2

Le groupe d'espace est le groupe mathématique représentant l'ensemble des symétries de la structure crisatlline; outre les symétries ponctuelles du cristal décrites par le groupe ponctuel il possède en plus les symétries de translation du réseau.

In [3]:
 #Génère le groupe d'espace et affiche son symbole son numéro et son ordre
symb=Sym.get_space_group_symbol()
Spacegr=SpaceGroup(symb)  

print('Symbole: ',symb)
print('Numéro: ',Sym.get_space_group_number())
print('Ordre: ',Spacegr.order)
   

Symbole:  R-3m
Numéro:  166
Ordre:  36


Le groupe d'espace peut être obtenu à partir du type de maille et du groupe ponctuel (déjà étudiés dans le notebook [lattice](lattice.ipynb)). La maille primitive étant rhombohédrique et le groupe ponctuel 3̅m, le groupe d'espace est symbolisé par ***R3̅m*** et porte le numéro **166** dans les tables internationales. Ce groupe possède 36 éléments de symétries et comme le groupe ponctuel du CdBr2 est d'ordre 12 il y'a en plus 24 éléments correspondant à des symétries de translations dans le groupe d'espace. Ici sera étudié en particulier 3 élements: une roto-inversion, un plan miroir et une rotation hélicoïdale.

## Roto-inversion d'ordre 3


In [6]:
#1:Rotoinversion
Op1=Spacegr[7]
point1=[1/3,2/3,-1/3]
coord1=Op1.operate(point1)

print('Matrice associée: ')
print(Op1)
print()
print('Point de départ: ', point1)
print()
print('Point d arrivée: ',coord1)

Matrice associée: 
Rot:
[[ 0.  1.  0.]
 [-1.  1.  0.]
 [ 0.  0. -1.]]
tau
[0. 0. 0.]

Point de départ:  [0.3333333333333333, 0.6666666666666666, -0.3333333333333333]

Point d arrivée:  [0.66666667 0.33333333 0.33333333]


Cet élément de symétrie est le 4e du fichier CIF,il est caractérisé par un axe de rotation d'ordre 3 de direction $[0,0,1]$ et un centre d'inversion en $(0,0,0)$. Elle effectue sur un point de l'espace une rotation de $+120°$ autour de son axe accompagnée d'une inversion et a comme matrice de transformation:

$$S_4=
\left(\begin{array}{ccc} 
0 & 1& 0\\
-1 & 1 & 0\\
0& 0& -1
\end{array}\right)$$

Celle-ci est exprimé dans la base conventionnelle (hexagonale) du réseau et par cet élément, tout atome situé sur un site $(x,y,z)$ sera envoyé sur un site équivalent $(y,-x+y,-z)$. Plus spécifiquement si l'on considère l'atome de Cadmium situé en $(1/3,2/3,-1/3)$, il subira la transformation $(1/3,2/3,-1/3)\rightarrow (2/3,1/3,1/3)$. Celle-ci est illustrée ci-dessous.

|<img src="Images/rot.jpg" alt="drawing" width="400"/>|
|:---:|
|Roto-inversion de l'atome de Cd $(1/3,2/3,-1/3)\rightarrow (2/3,1/3,1/3)$ L'inversion est représentée par le vecteur turquoise suivi d'une rotation en rouge dont l'axe correspond au vecteur $C$  de la maille conventionnelle|

## Plan miroir $(\bar{2},1,0)$

In [4]:
 #2: Plan miroir 
Op2=Spacegr[11]
point2=[2/3,1/3,1/3]
coord2=Op2.operate(point2)

print('Matrice associée: ')
print(Op2)
print()
print('Point de départ: ', point2)
print()
print('Point d arrivée: ',coord2)

Matrice associée: 
Rot:
[[-1.  1.  0.]
 [ 0.  1.  0.]
 [ 0.  0.  1.]]
tau
[0. 0. 0.]

Point de départ:  [0.6666666666666666, 0.3333333333333333, 0.3333333333333333]

Point d arrivée:  [-0.33333333  0.33333333  0.33333333]


Cette symétrie miroir est la 10e dans le fichier CIF et peut être décrite par un plan miroir dont les indices de miller sont $(\bar{2},1,0)$ et a pour matrice de transformation:

$$S_{10}=
\left(\begin{array}{ccc} 
-1 & 1& 0\\
0 & 1 & 0\\
0& 0& 1
\end{array}\right)$$

Par cet élément, l'atome situé sur le site $(x,y,z)$ sera envoyé sur un site équivalent $(-x+y,y,z)$. En particulier avec l'atome de Cd se trouvant en $(2/3,1/3,1/3)$, celui-ci aura pour application  $(2/3,1/3,1/3)\rightarrow (-1/3,1/3,1/3)$ figurée dans le schéma correspondant.

|<img src="Images/miroir.jpg" alt="drawing" width="400"/>|
|:---:|
|Symétrie orthogonale de l'atome de Cd  $(2/3,1/3,1/3)\rightarrow (-1/3,1/3,1/3)$ représentée par le vecteur turquoise, le plan miroir est représenté en vert.|

## Axe hélicoïdale $3_1$

In [7]:
 #3: Axe hélicoidale
Op3=Op1.from_rotation_and_translation(rotation_matrix=((-1, 1, 0), (-1, 0, 0), (0, 0, 1)), translation_vec=(-1/3, 1/3, 1/3), tol=0.1)
point3=[-1/3,1/3,1/3]
coord3=Op3.operate(point3)

print('Matrice associée: ')
print(Op3)
print()
print('Point de départ: ', point3)
print()
print('Point d arrivée: ',coord3)
   

Matrice associée: 
Rot:
[[-1.  1.  0.]
 [-1.  0.  0.]
 [ 0.  0.  1.]]
tau
[-0.33333333  0.33333333  0.33333333]

Point de départ:  [-0.3333333333333333, 0.3333333333333333, 0.3333333333333333]

Point d arrivée:  [0.33333333 0.66666667 0.66666667]


17e symétrie dans le fichier CIF, elle est caractérisée par un axe parallèle au vecteur $[0,0,1]$ du réseau et qui passe par le point $(0,1/3,0)$. L'effet de cet élément est d'appliquer une rotation de $-120°$ autour de cet axe et une translation de vecteur $[0,0,1/3]$ (rotation hélicoïdale $3_1$ dans le sens horloger); mathématiquement celà s'exprime:

$$
\left(\begin{array}{ccc} 
-1 & 1& 0\\
-1 & 0 & 0\\
0& 0& 1
\end{array}\right)
.
\left(\begin{array}{c} 
x \\
y\\
z
\end{array}\right)
+
\left(\begin{array}{c} 
-1/3 \\
1/3\\
1/3
\end{array}\right)
=
\left(\begin{array}{c} 
-1/3-x+y \\
1/3-x\\
1/3+z
\end{array}\right)
$$



Ce qui est donc équivalent à une rotation autour de l'axe $C$ suivi d'une translation de vecteur $[-1/3,1/3,1/3]$. Un exemple de cette symétrie est schématisée en dessous; l'atome de Cd en $(-1/3,1/3,1/3)$ est envoyé sur le site correspondant $(1/3,2/3,2/3)$.

|<img src="Images/hel.jpg" alt="drawing" width="400"/>|
|:---:|
|Rotation hélicoïdale $3_1$ de Cd  $(-1/3,1/3,1/3)\rightarrow (1/3,2/3,1/3)$ avec la flèche orange représentant la translation $[0,0,1/3]$ et la rouge montre la rotation autour de l'axe|